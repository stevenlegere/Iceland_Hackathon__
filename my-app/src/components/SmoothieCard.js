import React from 'react';
import '../index.css';
// Interface defining the properties for the SmoothieCard component


// SmoothieCard component to display information about a smoothie
const SmoothieCard = ( { smoothie }) => {
    return (
        // Container for the smoothie card with the "smoothie-card" class
        <div className="smoothie-card">
            <img src={smoothie.image} alt="Smoothie" className='smoothie-image' />
            <h3>{smoothie.title}</h3>
            <p>{smoothie.allergens}</p>
            <div className="rating">{smoothie.rating}</div>
            <button className="add_to_cart"> Add to cart</button>
        </div>
    )
}

export default SmoothieCard;