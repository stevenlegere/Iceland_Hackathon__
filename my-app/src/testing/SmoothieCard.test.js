import React from 'react';
import { render } from '@testing-library/react';
import SmoothieCard from '../components/SmoothieCard';

// Test suite for the SmoothieCard component
describe('SmoothieCard', () => {

    // Test case to check that the SmoothieCard component renders correctly
    it('renders correctly', () => {
        // Render the SmoothieCard component
        const { container } = render(<SmoothieCard smoothie={{title: 'Smoothie', allergens: 'nuts', ingredients: 'banana, milk, nuts', rating: 4}} />);
        // Check that the SmoothieCard component renders correctly
        expect(container).toMatchSnapshot();
    });

    // Test case to check that the SmoothieCard component renders the correct smoothie title
    it('renders the correct smoothie title', () => {
        // Render the SmoothieCard component
        const { getByText } = render(<SmoothieCard smoothie={{title: 'Smoothie', allergens: 'nuts', ingredients: 'banana, milk, nuts', rating: 4}} />);
        // Check that the SmoothieCard component renders the correct smoothie title
        expect(getByText('Smoothie')).toBeInTheDocument();
    });

    // Test case to check that the SmoothieCard component renders the correct smoothie allergens
    it('renders the correct smoothie allergens', () => {
        // Render the SmoothieCard component
        const { getByText } = render(<SmoothieCard smoothie={{title: 'Smoothie', allergens: 'nuts', ingredients: 'banana, milk, nuts', rating: 4}} />);
        // Check that the SmoothieCard component renders the correct smoothie allergens
        expect(getByText('nuts')).toBeInTheDocument();
    });

    // Test case to check that the SmoothieCard component renders the correct smoothie rating
    it('renders the correct smoothie rating', () => {
        // Render the SmoothieCard component
        const { getByText } = render(<SmoothieCard smoothie={{title: 'Smoothie', allergens: 'nuts', ingredients: 'banana, milk, nuts', rating: 4}} />);
        // Check that the SmoothieCard component renders the correct smoothie rating
        expect(getByText('4')).toBeInTheDocument();
    });


});


