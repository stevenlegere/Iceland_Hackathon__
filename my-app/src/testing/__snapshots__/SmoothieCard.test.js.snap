// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmoothieCard renders correctly 1`] = `
<div>
  <div
    class="smoothie-card"
  >
    <img
      alt="Smoothie"
      class="smoothie-image"
    />
    <h3>
      Smoothie
    </h3>
    <p>
      nuts
    </p>
    <div
      class="rating"
    >
      4
    </div>
    <button
      class="add_to_cart"
    >
       Add to cart
    </button>
  </div>
</div>
`;
